components/layout_2020/flexbox/mod.rs

```diff
@@ -15,10 +26,162 @@ mod layout;
+        if self
+            .independent_formatting_context
+            .style()
+            .get_box()
+            .overflow_x
+            .is_scrollable()
+        {
+            return Au::zero();
+        }
+
+        if cross_axis_is_item_block_axis {
+            let specified_size_suggestion = content_box_size.inline;
+
+            let transferred_size_suggestion = match self.independent_formatting_context {
+                IndependentFormattingContext::NonReplaced(_) => None,
+                IndependentFormattingContext::Replaced(ref bfc) => {
+                    match (
+                        bfc.contents.inline_size_over_block_size_intrinsic_ratio(
+                            self.independent_formatting_context.style(),
+                        ),
+                        content_box_size.block,
+                    ) {
+                        (Some(ratio), AuOrAuto::LengthPercentage(block_size)) => {
+                            let block_size = block_size.clamp_between_extremums(
+                                min_size.block.auto_is(Au::zero),
+                                max_size.block,
+                            );
+                            Some(block_size.scale_by(ratio))
+                        },
+                        _ => None,
+                    }
+                },
+            };
+
+            let inline_content_size = self
+                .independent_formatting_context
+                .inline_content_sizes(layout_context)
+                .min_content;
+            let content_size_suggestion = match self.independent_formatting_context {
+                IndependentFormattingContext::NonReplaced(_) => inline_content_size,
+                IndependentFormattingContext::Replaced(ref replaced) => {
+                    if let Some(ratio) = replaced
+                        .contents
+                        .inline_size_over_block_size_intrinsic_ratio(
+                            self.independent_formatting_context.style(),
+                        )
+                    {
+                        inline_content_size.clamp_between_extremums(
+                            min_size.block.auto_is(Au::zero).scale_by(ratio),
+                            max_size.block.map(|l| l.scale_by(ratio)),
+                        )
+                    } else {
+                        inline_content_size
+                    }
+                },
+            };
+
+            let result = match specified_size_suggestion {
+                AuOrAuto::LengthPercentage(l) => l.min(content_size_suggestion),
+                AuOrAuto::Auto => {
+                    if let Some(l) = transferred_size_suggestion {
+                        l.min(content_size_suggestion)
+                    } else {
+                        content_size_suggestion
+                    }
+                },
+            };
+            result.clamp_below_max(max_size.inline)
```

Comment:
This seems wrong. Min size usually takes precedence over max size. Perhaps this should only apply if the min size is a computed intrinsic size rather an actual `min-` style?

Reply:
It's possible that is wrong, but this code comes from the automatic min-size computation that comes from before this change. This change reuses the computation, but fixing it is probably a separate issue.

Reply:
I have taken a deeper looker at whether or not we should be clamping here and the [specification](https://drafts.csswg.org/css-flexbox/#min-size-auto) says that we should be:

> The content-based minimum size of a [flex item](https://drafts.csswg.org/css-flexbox/#flex-item) is the smaller of its [specified size suggestion](https://drafts.csswg.org/css-flexbox/#specified-size-suggestion) and its [content size suggestion](https://drafts.csswg.org/css-flexbox/#content-size-suggestion) if its specified size suggestion exists; otherwise, the smaller of its [transferred size suggestion](https://drafts.csswg.org/css-flexbox/#transferred-size-suggestion) and its content size suggestion if the element is [replaced](https://drafts.csswg.org/css-display-4/#replaced-element) and its transferred size suggestion exists; otherwise its content size suggestion. In all cases, the size is clamped by the [maximum](https://drafts.csswg.org/css-sizing-3/#max-width) [main size](https://drafts.csswg.org/css-flexbox/#main-size) if it’s [definite](https://drafts.csswg.org/css-flexbox/#definite).